### ISSUE TITLE
Past search should remember individual parts of the query.

### ISSUE THREAD

**Issue by Pauan** [Pauan/tab-organizer#56]
Originally reported on Google Code with ID 55

```
What steps will reproduce the problem?
1. Type in a query like "foo"
2. Later on, type in a query like "foo OR bar"
3. Type in "foo"

What is the expected output? What do you see instead?
It should continue to show "foo" on the dropdown box. Instead, it shows "foo OR bar",
and typing in "bar" shows nothing.

Please provide any additional information below.
Basically, when using advanced syntax (mostly OR and AND), it should save each individual
part, rather than the entire query.
```

Reported by `pcxunlimited` on 2010-09-25 03:27:08

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 22:16:41
- **Labels added**: Milestone-5

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:03:36
- **Labels added**: Type-Internal
- **Labels removed**: Type-Bug

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:14:51
- **Labels removed**: Usability

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:17:41
- **Labels removed**: OS-All

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:18:37
- **Labels removed**: Component-Search

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:39:18
- **Labels removed**: Priority-Low

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:47:28
- **Status changed**: `Assigned`
